### PR TITLE
Convert executor/restore_test.go to unit test

### PR DESCRIPTION
**Pull request by cimnine**
## Summary

`executor/restore_test.go` was basically a huge unit test all along. It has now been taken apart and re-implemented as an actual unit test in preparations for #274.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`,
      as they show up in the changelog
- [x] Update the documentation.
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

File: executor/restore_test.go [hunk 1]
```diff
@@ -2,306 +2,270 @@ package executor
...
 }
 
-func newConfig() *job.Config {
-	cfg := job.NewConfig(context.TODO(), nil, nil, &k8upv1alpha1.Restore{}, scheme, "")
-	return &cfg
+func assertVolumes(t *testing.T, ex PVCExpectation, volumes []corev1.Volume) {
+	if !ex.Expected {
+		assert.Len(t, volumes, 0)
+		return
+	}
```
Comment (ccremer) [inline executor/restore_test.go:223]: I don't see any tests now that use this flag. Maybe we can remove it?

File: executor/restore_test.go [hunk 2]
```diff
@@ -2,306 +2,270 @@ package executor
...
+			e := NewRestoreExecutor(*tt.GivenJobConfig)
+			volumes, mounts := e.volumeConfig(tt.GivenResource)
+
+			assertVolumes(t, tt.ExpectedPVC, volumes)
+			assertVolumeMonts(t, tt.ExpectedVolumeMount, mounts)
 		})
 	}
+}
+
+func assertVolumeMonts(t *testing.T, ex VolumeMountExpectation, mounts []corev1.VolumeMount) {
```
Comment (ccremer) [inline executor/restore_test.go:206]:
```suggestion
func assertVolumeMounts(t *testing.T, ex VolumeMountExpectation, mounts []corev1.VolumeMount) {
```

File: executor/restore_test.go [hunk 3]
```diff
@@ -2,306 +2,270 @@ package executor
...
+			ExpectedPVC: PVCExpectation{
+				Expected:  true,
+				ClaimName: "test",
+				ReadOnly:  false,
+			},
+			ExpectedVolumeMount: VolumeMountExpectation{
+				Expected: true,
+				Name:     "test",
+				Path:     "/restore",
+			},
```
Comment (ccremer) [inline executor/restore_test.go:191]: Maybe we can even remove the struct and assert directly in the test?
```suggestion
			ExpectedPVCClaimName: "test",
			ExpectedVolumeMountName: "test",
			ExpectedVolumeMountPath: "/restore"
```

**Review by ccremer — COMMENTED** [review]
I like it! The tests are now much more readable

File: executor/restore_test.go (hunk 1 above)
Comment (cimnine) [reply executor/restore_test.go:223]: `givenS3RestoreResource_whenVolumeConfig_expectNoPVCAndNoMount` implicitly sets them `false`. I've now made this explicit.

File: executor/restore_test.go (hunk 3 above)
Comment (cimnine) [reply executor/restore_test.go:193]: I find the structure cleaner, especially when passing them to `assertVolumeMounts` and `assertVolumes`.

**Comment by cimnine** [comment]
I've implemented the feedback. There have not been any other changes.

**Review by ccremer — APPROVED** [review]

**Pull request merged**